File: src/client/java/com/somefrills/events/MouseClickEvent.java
package com.somefrills.events;

import com.somefrills.misc.KeyAction;
import net.minecraft.client.gui.Click;
import net.minecraft.client.input.MouseInput;

public class MouseClickEvent extends Cancellable {
    public Click click;
    public MouseInput input;
    public KeyAction action;

    public MouseClickEvent(Click click, KeyAction action) {
        this.click = click;
        this.input = click.buttonInfo();
        this.action = action;
    }

    public int button() {
        return this.input.button();
    }
}

File: src/main/kotlin/com/somefrills/events/MouseClickEvent.kt
package com.somefrills.events

import com.somefrills.misc.KeyAction
import net.minecraft.client.input.MouseButtonEvent
import com.somefrills.events.FrillsEvent.Cancellable

class MouseClickEvent(val click: MouseButtonEvent, @JvmField val action: KeyAction) : FrillsEvent(), Cancellable {
    fun button(): Int {
        return click.button()
    }
}
